server/pulp/server/controllers/repository.py

```diff
@@ -105,6 +105,46 @@ def find_repo_content_units(
             yield_count += 1
 
 
+def find_units_not_downloaded(repository):
+    """
+    Find content units that have not been fully downloaded.
+
+    :param repository: A repository.
+    :type repository: model.Repository
+    :return: The requested units.
+    :rtype: generator
+    """
+    types = {}
+    for association in model.RepositoryContentUnit.objects(repo_id=repository.repo_id):
```

Comment:
I think it's reasonable to use ``.only('unit_id')`` here to avoid loading extra data about every association.